### PR TITLE
Fix/overlay bugs

### PR DESCRIPTION
**Pull request by tlouisse**
- closes https://github.com/ing-bank/lion/issues/817
- closes https://github.com/ing-bank/lion/issues/816
- closes https://github.com/ing-bank/lion/issues/803

### CODE REVIEW

File: packages/overlays/src/OverlayController.js [hunk 1]
```diff
@@ -358,11 +357,14 @@ export class OverlayController {
   __setupTeardownAccessibility({ phase }) {
     if (phase === 'init') {
       this.__storeOriginalAttrs(this.contentNode, ['role', 'id']);
```
Comment (jorenbroekema) [inline packages/overlays/src/OverlayController.js:359]: Should we guard this.contentNode as well?

File: packages/overlays/test/OverlayController.test.js [hunk 2]
```diff
@@ -430,6 +430,8 @@ describe('OverlayController', () => {
         // Don't hide on inside (content) click
         ctrl.contentNode.click();
         await aTimeout();
```
Comment (jorenbroekema) [inline packages/overlays/test/OverlayController.test.js:432]: These two timeouts are necessary for the double setTimeout in the src? So if we removed these timeouts here in the test, and revert the fix, this test would pass, there we keep the timeouts in this test so it's a proper regression test? :) Just for my understanding

File: packages/overlays/src/OverlayController.js (hunk 1 above)
Comment (tlouisse) [reply packages/overlays/src/OverlayController.js:359]: It already is: https://github.com/ing-bank/lion/blob/master/packages/overlays/src/OverlayController.js#L249

File: packages/overlays/test/OverlayController.test.js (hunk 2 above)
Comment (tlouisse) [reply packages/overlays/test/OverlayController.test.js:433]: They are not neccessary (anymore), removed them 👍

**Pull request merged**